sql/core/src/main/scala/org/apache/spark/sql/execution/python/FlatMapGroupsInPandasExec.scala

```diff
@@ -89,7 +89,8 @@ case class FlatMapGroupsInPandasExec(
         Array(argOffsets),
```

Comment:
e.g., define `val localPythonMetrics = pythonMetrics` at line 79, and replace `pythonMetrics` to `localPythonMetrics`.

Reply:
Thanks @HyukjinKwon for looking into this.
Unfortunately the proposed solution/test of using `val localPythonMetrics = pythonMetrics` does not appear to work.
Using lazy val for the metrics appears to break many tesdts with Python. In particular I can see in that case that when using pyspark and "going through rdd" as in  `df_with_udf.rdd.collect()` we get java.lang.NullPointerException.  
I would not propose to skip the failing test in postgreSQL/udf-aggregates_part3.sql, but rather move it to a Python test: see test_pandas_udf_nested in test_pandas_udf.py
However if we can understand more clearly where this issue comes from, all the better.

Reply:
I'd like to clarify that the proposed solution for the failing test is to move the test to a Python test in python/pyspark/sql/tests/test_pandas_udf.py where it runs OK. I would (try to) argue that the issue may come for the way the Python UDF test in udf-aggregates_part3.sql is executed via Scala and the fact that that particular test is referring to `udf` twice: `udf((select udf(count(*))` which apparently creates an issue there, while it works fine in a Python test.